src/main/java/org/jabref/gui/commonfxcontrols/FilenamePatternItemModel.java

```diff
@@ -0,0 +1,47 @@
+package org.jabref.gui.commonfxcontrols;
+
+import java.util.Objects;
+
+import javafx.beans.property.ObjectProperty;
+import javafx.beans.property.SimpleObjectProperty;
+import javafx.beans.property.SimpleStringProperty;
+import javafx.beans.property.StringProperty;
+
+import org.jabref.model.entry.types.EntryType;
+
+public class FilenamePatternItemModel {
+    private final ObjectProperty<EntryType> entryType = new SimpleObjectProperty<>();
+    private final StringProperty pattern = new SimpleStringProperty("");
+
+    public FilenamePatternItemModel(EntryType entryType, String pattern) {
+        Objects.requireNonNull(entryType);
+        Objects.requireNonNull(pattern);
+        this.entryType.setValue(entryType);
+        this.pattern.setValue(pattern);
+    }
+
+    public EntryType getEntryType() {
+        return entryType.getValue();
+    }
+
+    public ObjectProperty<EntryType> entryType() {
+        return entryType;
+    }
+
+    public void setPattern(String pattern) {
+        this.pattern.setValue(pattern);
+    }
+
+    public String getPattern() {
+        return pattern.getValue();
+    }
+
+    public StringProperty pattern() {
+        return pattern;
+    }
+
+    @Override
+    public String toString() {
+        return "[" + entryType.getValue().getName() + "," + pattern.getValue() + "]";
+    }
```

Comment:
I would put that in a separate method - `toString()` is more used for debugging.